distributed/client.py

```diff
@@ -1074,9 +1071,9 @@ def _close(self, fast=False):
                 pc.stop()
             self._scheduler_identity = {}
             with ignoring(AttributeError):
-                dask.config.set(scheduler=self._previous_scheduler)
-            with ignoring(AttributeError):
-                dask.config.set(shuffle=self._previous_shuffle)
+                # clear the dask.config set keys
+                with self._set_config:
+                    pass
             if self.get == dask.config.get('get', None):
                 del dask.config.config['get']
```

Comment:
What if the user had specified something prior to starting the scheduler, like 

```python
dask.config.set(scheduler='threads')
with Client():
    pass
assert dask.config.get('scheduler') == 'threads'
```

Reply:
Same comment with shuffle.

Reply:
Just tested it, the assertion passes; the `with self._set_config` restores the previous settings at exit.

Reply:
OK, you're right.  That's a nice solution.